<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE beans PUBLIC "-//SPRING//DTD BEAN 2.0//EN" "https://www.springframework.org/dtd/spring-beans-2.0.dtd">

<!--
	- 自定义异常错误码
	-->
<beans>
    <!--设置 H2 数据库相关异常-->
    <bean id="H2" class="org.springframework.jdbc.support.SQLErrorCodes">
        <property name="badSqlGrammarCodes">
            <value>42000,42001,42101,42102,42111,42112,42121,42122,42132</value>
        </property>
        <property name="duplicateKeyCodes">
            <value>23001,23505</value>
        </property>
        <property name="dataIntegrityViolationCodes">
            <value>22001,22003,22012,22018,22025,23000,23002,23003,23502,23503,23506,23507,23513</value>
        </property>
        <property name="dataAccessResourceFailureCodes">
            <value>90046,90100,90117,90121,90126</value>
        </property>
        <property name="cannotAcquireLockCodes">
            <value>50200</value>
        </property>
        <!--设置自定义异常转换器-->
        <!--// Look for defined custom translations first-->
        <!--当设置自定义异常时,会优先匹配自定义异常-->
        <property name="customTranslations">
            <!--设置当前自定义sql异常定义转换器-->
            <bean class="org.springframework.jdbc.support.CustomSQLErrorCodesTranslation">
                <!--设置拦截的自定义错误码-->
                <property name="errorCodes">
                    <value>23001,23505</value>
                </property>
                <!--设置拦截到错误码后抛出的异常类型-->
                <property name="exceptionClass">
                    <value>com.xingguo.spring.jdbc.exception.CustomDuplicateKeyException</value>
                </property>
            </bean>
        </property>
    </bean>

</beans>
